app/helpers/writable_helper.rb

```diff
@@ -63,4 +63,11 @@ def shortened_desc(desc, id)
       content_tag(:span, sanitize_simple_link_text(desc[255..-1]), class: 'hidden', id: "desc-#{id}") +
       content_tag(:a, 'more &raquo;'.html_safe, href: '#', id: "expanddesc-#{id}", class: 'expanddesc')
   end
+
+  def allowed_boards(post, user)
+    valid_ids = BoardAuthor.where(user: user).pluck(:board_id)
+    valid_ids << post.board_id if post.board_id.present?
```

Comment:
Is this valid? How would you get into this state and why is it permitted?

Reply:
Mod edits of posts in boards they can't post in, iirc